gammapy/datasets/make.py

```diff
@@ -64,3 +72,146 @@ def sigma_energy_theta(energy, theta, sigma):
+    # time for the last run to finish
+    night_start = Quantity(22., 'hour')
+    night_duration = Quantity(5.5, 'hour')
+    hour_start = night_start + night_duration * np.random.random(len(obs_id))
+
+    # add night hour to integer part of MJD
+    time_start += hour_start
+
+    if debug:
+        # show the observation times in UTC
+        time_start = time_start.iso
+    else:
+        # show the observation times in seconds after the reference
+        time_start = time_relative_to_ref(time_start, header)
+        # converting to quantity (beter treatment of units)
+        time_start = Quantity(time_start.sec, 'second')
+
+    obs_table['TIME_START'] = time_start
+
+    # stop time
+    # calculated as TIME_START + TIME_OBSERVATION
+    if debug:
+        time_stop = Time(obs_table['TIME_START']) + \
+            TimeDelta(obs_table['TIME_OBSERVATION'])
+    else:
+        time_stop = TimeDelta(
+            obs_table['TIME_START']) + TimeDelta(obs_table['TIME_OBSERVATION'])
+        # converting to quantity (beter treatment of units)
+        time_stop = Quantity(time_stop.sec, 'second')
+
+    obs_table['TIME_STOP'] = time_stop
+
+    # az, alt
+    # random points in a sphere above 45 deg altitude
+    az, alt = sample_sphere(len(obs_id), (0, 360), (45, 90), 'degree')
+    az = Angle(az, 'degree')
+    alt = Angle(alt, 'degree')
+    obs_table['AZ'] = az
+    obs_table['ALT'] = alt
+
+    # RA, dec
+    # derive from az, alt taking into account that alt, az represent the values
+    # at the middle of the observation, i.e. at time_ref + (TIME_START + TIME_STOP)/2
+    # (or better: time_ref + TIME_START + (TIME_OBSERVATION/2))
+    # in debug modus, the time_ref should not be added, since it's already included
+    # in TIME_START and TIME_STOP
+    az = Angle(obs_table['AZ'])
+    alt = Angle(obs_table['ALT'])
+    if debug:
+        obstime = Time(obs_table['TIME_START']) + \
+            TimeDelta(obs_table['TIME_OBSERVATION']) / 2.
+    else:
+        obstime = time_ref_from_dict(obs_table.meta) + TimeDelta(
+            obs_table['TIME_START']) + TimeDelta(obs_table['TIME_OBSERVATION']) / 2.
+    location = observatory_locations[observatory_name]
+    alt_az_coord = AltAz(az=az, alt=alt, obstime=obstime, location=location)
+    # optional: make it depend on other pars: temperature, pressure,
+    # humidity,...
+    sky_coord = alt_az_coord.transform_to(FK5)
+    obs_table['RA'] = sky_coord.ra
+    obs_table['DEC'] = sky_coord.dec
+
+    # positions
+
+    # number of telescopes
+    # random integers between 3 and 4
+    n_tels_min = 3
+    n_tels_max = 4
+    n_tels = np.random.randint(n_tels_min, n_tels_max + 1, len(obs_id))
+    obs_table['N_TELS'] = n_tels
+
+    # muon efficiency
+    # random between 0.6 and 1.0
+    muon_efficiency_min = 0.6
+    muon_efficiency_max = 1.0
+    muon_efficiency = np.random.random(
+        len(obs_id)) * (muon_efficiency_max - muon_efficiency_min) + muon_efficiency_min
```

Comment:
I also don't, but I also don't like breaking a simple expression into many lines of code with temporary variables: it makes the code very hard to read. This was the suggestion of `autopep8` or long lines of code. I'd rather have it like this instead of with many intermediate steps.